core/domain/summary_services_test.py

```diff
@@ -336,28 +338,28 @@ def test_for_featured_explorations(self):
         The call to get_featured_explorations() should only return
         [EXP_ID_2].
         """
+        activity_services.update_featured_activity_references([
+            activity_domain.ActivityReference(
+                feconf.ACTIVITY_TYPE_EXPLORATION, self.EXP_ID_2)
+        ])
 
-        rights_manager.publicize_exploration(self.admin_id, self.EXP_ID_2)
-
-        featured_exploration_summaries = (
-            summary_services.get_featured_exploration_summary_dicts([
+        featured_activity_summaries = (
```

Comment:
Do you test for ensuring the language code filter works correctly? Might be worth adding another test for this. What happens if the language code list is empty?